### PR TITLE
base: Use requirejs as our AMD loader

### PR DESCRIPTION
**Pull request by stefwalter**
This uses requirejs as our AMD loader, and drops our custom implementation. Most instances of cockpit.js becomes AMD loaded rather than loaded in a script tag.

### CODE REVIEW

File: test/check-dashboard [hunk 1]
```diff
@@ -141,9 +141,9 @@ class TestDashboard(MachineCase):
         b.leave_page()
         b.wait_js_cond('$("#machine-dropdown ul li").length == 2')
         b.click('#machine-dropdown ul li:nth-child(1) a')
-        b.wait_js_cond('cockpit.location.href == "/@localhost"')
+        b.wait_js_cond('window.location.hash == "#/@localhost"')
```
Comment (mvollmer) [inline test/check-dashboard]: I don't understand this change.  Is it at all related to require.js?

Comment (stefwalter) [inline test/check-dashboard]: 'cockpit' is only valid inside the private scope ... it's no longer a global on pages that use AMD ... such as the index page.

Comment (mvollmer) [inline test/check-dashboard]: Right, makes sense.

**Comment by stefwalter** [comment]
Not sure how the failure relates to the change. Suspect a false possitive in hubbot.

**Comment by mvollmer** [comment]
The color chooser popup does indeed not close anymore.  How obscure is that?  I'll investigate...

**Comment by mvollmer** [comment]
> The color chooser popup does indeed not close anymore during manual testing. How obscure is that? I'll investigate...

Right, this happens because bootstrap is now loaded twice into the shell.

**Comment by stefwalter** [comment]
Fixed issue with loading jquery twice.

**Pull request closed without merging**